### PR TITLE
feat: publish databse to Web

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

added support for publishing one or multiple databases. 

Screenshot 2024-07-12 at 15 21 20

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/.vscode/launch.json [hunk 1]
```diff
@@ -116,11 +116,11 @@
...
             "name": "AF-desktop: Debug Rust",
             "type": "lldb",
-            "request": "attach",
-            "pid": "${command:pickMyProcess}"
+            // "request": "attach",
+            // "pid": "${command:pickMyProcess}"
             // To launch the application directly, use the following configuration:
-            // "request": "launch",
-            // "program": "[YOUR_APPLICATION_PATH]",
+            "request": "launch",
```
Comment (LucasXu0) [inline frontend/.vscode/launch.json]: revert the code

File: frontend/appflowy_flutter/integration_test/desktop/uncategorized/share_markdown_test.dart [hunk 2]
```diff
@@ -105,12 +104,12 @@ fn main() {
 > Click `?` at the bottom right for help and support.
 
 > 🥰
-> 
+>
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/integration_test/desktop/uncategorized/share_markdown_test.dart]: keep the space here

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/built_in_page_widget.dart [hunk 3]
```diff
@@ -141,7 +138,7 @@ class _BuiltInPageWidgetState extends State<BuiltInPageWidget> {
             width: 24,
             height: 24,
             iconPadding: const EdgeInsets.all(3),
-            iconColorOnHover: Theme.of(context).colorScheme.onSecondary,
+            // iconColorOnHover: Theme.of(context).colorScheme.onSecondary,
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/base/built_in_page_widget.dart]: remove the code

File: frontend/appflowy_flutter/lib/plugins/shared/share/share_bloc.dart [hunk 4]
```diff
@@ -2,42 +2,42 @@ import 'dart:io';
...
 import 'package:appflowy_backend/protobuf/flowy-user/protobuf.dart';
 import 'package:appflowy_result/appflowy_result.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 
-part 'document_share_bloc.freezed.dart';
+part 'share_bloc.freezed.dart';
 
-const _url = 'https://appflowy.com';
+const _url = 'https://beta.appflowy.com';
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/plugins/shared/share/share_bloc.dart:18]: revert this change

File: frontend/rust-lib/flowy-folder/src/manager.rs [hunk 5]
```diff
@@ -1182,24 +1209,25 @@ impl FolderManager {
 
     Some(view)
   }
+
   async fn get_publish_payload(
```
Comment (appflowy) [inline frontend/rust-lib/flowy-folder/src/manager.rs:1213]: it would be better to yield at some point if this task is a cpu bound or using spawn_blocking instaed

for example, If you don't use spawn_blocking, you need to yield by using `tokio::task::yield_now().await;` Otherwise, you might stuck the runtime

File: frontend/rust-lib/flowy-core/src/deps_resolve/folder_deps.rs [hunk 6]
```diff
@@ -300,13 +297,31 @@ impl FolderOperationHandler for DatabaseFolderOperation {
...
-    _view_id: &str,
-    _layout: ViewLayout,
-  ) -> FutureResult<EncodedCollab, FlowyError> {
-    // Database view doesn't support collab
-    FutureResult::new(async move { Err(FlowyError::not_support()) })
+  fn get_encoded_collab_v1(&self, view_id: &str) -> FutureResult<EncodedCollabWrapper, FlowyError> {
+    let database_manager = self.0.clone();
+    let view_id = view_id.to_string();
+    FutureResult::new(async move {
+      let database_encoded_collab = database_manager
```
Comment (appflowy) [inline frontend/rust-lib/flowy-core/src/deps_resolve/folder_deps.rs:304]: You can directly read data from rocksdb. using database_manager will cause some unnecessary logic. For example, it will trigger sync for everying rows and database
